### ISSUE TITLE
New tests now running and failing on vm-kernel-reload builders

### ISSUE THREAD

**Issue by whesse** [dart-lang/sdk#37905]
TLDR; Some tests with unwanted (unexpected) compile-time errors are no longer skipped on vm-kernel-reload builders.

We skip tests with expected compile-time errors on the reload bots, since they should never get to the reload phase anyway.  The test for expected compile-time errors was looking at both the annotations in the test (the good, new way, especially for multitests) and at the old status files.

We are removing all but the Skip and Slow and Crash lines from the status files, so these tests are now expected to pass, and no longer skipped.  They are failing on reload bots, just as they do on the other vm bots.  So I am approving them there, just like they are already approved on the other vm bots.  They are currently failing with compile-time error on dartkb and kernel-precomp builders.

I think this issue can be closed, or marked as duplicate of any issue already filed about these tests.

| BOT                                         | TEST                                                    | RESULT           | EXPECTED | 
| ------------------------------------------- | ------------------------------------------------------- | ---------------- | -------- | 
| vm-kernel-reload-linux-release-x64          | language_2/cyclic_type2_test                            | CompileTimeError | Pass     |
| vm-kernel-reload-linux-release-x64          | language_2/cyclic_type_test/02                          | CompileTimeError | Pass     |
| vm-kernel-reload-linux-release-x64          | language_2/cyclic_type_test/04                          | CompileTimeError | Pass     |
| vm-kernel-reload-linux-release-x64          | language_2/dynamic_prefix_core_test/none                | CompileTimeError | Pass     |
| vm-kernel-reload-linux-release-x64          | language_2/generic_tearoff_test                         | CompileTimeError | Pass     |
| vm-kernel-reload-linux-release-x64          | language_2/instantiate_tearoff_of_call_test             | CompileTimeError | Pass     |
| vm-kernel-reload-linux-release-x64          | language_2/issue31596_super_test/01                     | CompileTimeError | Pass     |
| vm-kernel-reload-linux-release-x64          | language_2/issue31596_super_test/03                     | CompileTimeError | Pass     |
| vm-kernel-reload-linux-release-x64          | language_2/least_upper_bound_expansive_test/none        | CompileTimeError | Pass     |
| vm-kernel-reload-linux-release-x64          | lib_2/mirrors/deferred_type_test                        | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-debug-x64   | language_2/cyclic_type2_test                            | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-debug-x64   | language_2/cyclic_type_test/02                          | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-debug-x64   | language_2/cyclic_type_test/04                          | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-debug-x64   | language_2/dynamic_prefix_core_test/none                | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-debug-x64   | language_2/generic_tearoff_test                         | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-debug-x64   | language_2/instantiate_tearoff_of_call_test             | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-debug-x64   | language_2/issue31596_super_test/01                     | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-debug-x64   | language_2/issue31596_super_test/03                     | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-debug-x64   | language_2/least_upper_bound_expansive_test/none        | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-debug-x64   | lib_2/mirrors/deferred_type_test                        | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-release-x64 | language_2/cyclic_type2_test                            | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-release-x64 | language_2/cyclic_type_test/02                          | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-release-x64 | language_2/cyclic_type_test/04                          | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-release-x64 | language_2/dynamic_prefix_core_test/none                | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-release-x64 | language_2/generic_tearoff_test                         | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-release-x64 | language_2/instantiate_tearoff_of_call_test             | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-release-x64 | language_2/issue31596_super_test/01                     | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-release-x64 | language_2/issue31596_super_test/03                     | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-release-x64 | language_2/least_upper_bound_expansive_test/none        | CompileTimeError | Pass     |
| vm-kernel-reload-rollback-linux-release-x64 | lib_2/mirrors/deferred_type_test                        | CompileTimeError | Pass     |

**Comment by a-siva** [comment]
This doesn't appear to be a reload related issue but more of an unexpected compile time error issue (per comment this failure is seen in other bots too). Removing vm-hot-reload label and adding area-frontend label
